### PR TITLE
[Snackbar] Added API to make snackbar messages keep focus until an action is taken.

### PR DESCRIPTION
**Pull request by randallli**
Closes: #5211 

based off of cl/214202298

### CODE REVIEW

File: components/Snackbar/tests/unit/MDCSnackbarMessageViewTests.m [hunk 1]
```diff
@@ -91,4 +101,32 @@ - (void)testAccessibilityHintSetFromSnackbarMessageProperty {
...
+  self.manager.shouldSetAccessibilityViewIsModal = YES;
+
+  // When
+  [self.manager showMessage:self.message];
+
+  // Then
+  XCTAssertFalse(self.manager.internalManager.overlayView.accessibilityViewIsModal);
+}
+
+
```
Comment (jverkoey) [inline components/Snackbar/tests/unit/MDCSnackbarMessageViewTests.m]: Suggestion: we may want a test that confirms that `shouldSetAccessibilityViewIsModal` set to NO results in `accessibilityViewIsModal` being NO as well.

File: components/Snackbar/src/MDCSnackbarManager.h [hunk 2]
```diff
@@ -221,6 +221,12 @@
 @property(nonatomic, readwrite, setter=mdc_setAdjustsFontForContentSizeCategory:)
     BOOL mdc_adjustsFontForContentSizeCategory;
 
+/**
+ If enabled, accessibilityViewIsModal will be set for all non-transient snackbar views.
+
+ Default is set to NO.
+ */
+@property(nonatomic, assign) BOOL shouldSetAccessibilityViewIsModal;
```
Comment (jverkoey) [inline components/Snackbar/src/MDCSnackbarManager.h]: Suggestion: `shouldEnableAccessibilityViewIsModal`

`set` doesn't necessarily imply that we're going to enable the property.

**Review by jverkoey — APPROVED** [review]

File: components/Snackbar/src/MDCSnackbarManager.h [hunk 3]
```diff
@@ -221,6 +221,12 @@
 @property(nonatomic, readwrite, setter=mdc_setAdjustsFontForContentSizeCategory:)
     BOOL mdc_adjustsFontForContentSizeCategory;
 
+/**
+ If enabled, accessibilityViewIsModal will be set for all non-transient snackbar views.
```
Comment (jverkoey) [inline components/Snackbar/src/MDCSnackbarManager.h]: will be `enabled`

File: components/Snackbar/src/MDCSnackbarManager.m [hunk 4]
```diff
@@ -20,6 +20,7 @@
 #import "private/MDCSnackbarMessageInternal.h"
 #import "private/MDCSnackbarMessageViewInternal.h"
 #import "private/MDCSnackbarOverlayView.h"
+#import "private/MDCSnackbarManagerInternal.h"
```
[the remark this answers is not in the document]
Comment (jverkoey) [reply components/Snackbar/src/MDCSnackbarManager.m]: Clang's trying to tell you to sort the imports.

[the remark this answers is not in the document]
Comment (randallli) [reply components/Snackbar/src/MDCSnackbarManager.m]: if clang-format is suggesting that I delete this line, I cannot. It wont build if i dont have this.

[the remark this answers is not in the document]
Comment (jverkoey) [reply components/Snackbar/src/MDCSnackbarManager.m]: It's not - clang's trying to tell you that you need to sort your imports.

[the remark this answers is not in the document]
Comment (randallli) [reply components/Snackbar/src/MDCSnackbarManager.m]: ah ok thanks for explaining.

**Review by yarneo — APPROVED** [review]

**Pull request merged**